### ISSUE TITLE
'shared' is unavailable: Use view controller based solutions where appropriate instead. On public func isDisplayedInScreen

### ISSUE THREAD

**Issue by danieladarve** [kirualex/SwiftyGif#20]
Hello,

I am getting the following error when load the library into my project:

Only happens when building an extension, any ideas?

../UIImageView+SwiftyGif.swift:262:71: 'shared' is unavailable: Use view controller based solutions where appropriate instead.

```

public func isDisplayedInScreen(_ imageView:UIView?) ->Bool{
        if (self.isHidden) {
            return false
        }
        
      let screenRect = UIScreen.main.bounds
        let viewRect = imageView!.convert(self.frame,to:UIApplication.shared.keyWindow)
        
        let intersectionRect = viewRect.intersection(screenRect);
        if (intersectionRect.isEmpty || intersectionRect.isNull) {
            return false
        }
        return (self.window != nil)
}
```

**Comment by alexiscreuzot** [comment]
Are you using GIF images into your extension? Maybe trying to exclude this pod from your extension target would do the trick.

**Comment by danieladarve** [comment]
Yes, I'm using GIF images in the extension

**Comment by danieladarve** [comment]
I ended up modifying the function to make it work as:
```
public func isDisplayedInScreen(_ imageView:UIView?) ->Bool{
        if (self.isHidden) {
            return false
        }
        return (self.window != nil)
    }
```

**Comment by facumenzella** [comment]
You just have to change the invocation parameter
```swift 
let viewRect = imageView!.convert(self.frame, to:nil)
```
Passing nil is the same as passing UIApplication.shared.keyWindow
Like this...
```swift
public func isDisplayedInScreen(_ imageView:UIView?) ->Bool{
        if (self.isHidden) {
            return false
        }
        
        let screenRect = UIScreen.main.bounds
        let viewRect = imageView!.convert(self.frame, to:nil)
        
        let intersectionRect = viewRect.intersection(screenRect);
        if (intersectionRect.isEmpty || intersectionRect.isNull) {
            return false
        }
        return (self.window != nil)
    }
```

**Comment by neobeppe** [comment]
fixed in 3.0.2, thanks @facumenzella

**Comment by facumenzella** [comment]
![10188131-138350188_39-s4-v1](https://cloud.githubusercontent.com/assets/1125252/23704358/3ff7c0b0-03e3-11e7-8841-dcb895201202.png)
Thank you!
